presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -240,13 +242,21 @@ private Estimate summarizePartitionStatistics(
         }
 
         if (unpartitioned) {
-            return ImmutableMap.of(HivePartition.UNPARTITIONED_ID, getTableStatistics(tableHandle.getSchemaTableName(), tableColumns));
+            return ImmutableMap.of(HivePartition.UNPARTITIONED_ID, getTableStatistics(tableHandle.getSchemaTableName(), tableColumns.keySet()));
         }
         else {
-            return getPartitionsStatistics(tableHandle.getSchemaTableName(), hivePartitions, tableColumns);
+            return getPartitionsStatistics(tableHandle.getSchemaTableName(), hivePartitions, listNonPartitioningColumns(tableColumns, hivePartitions));
         }
     }
 
+    private static Set<String> listNonPartitioningColumns(Map<String, ColumnHandle> tableColumns, List<HivePartition> hivePartitions)
```

Comment:
You were supposed to remote `hivePartitions` arg.

Reply:
right, forgot about it